rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/telemetry/TbMsgAttributesNode.java

```diff
@@ -166,11 +171,34 @@ public TbPair<Boolean, JsonNode> upgrade(int fromVersion, JsonNode oldConfigurat
                     hasChanges = true;
                     ((ObjectNode) oldConfiguration).put(UPDATE_ATTRIBUTES_ONLY_ON_VALUE_CHANGE_KEY, false);
                 }
+            case 1:
+                // update notifyDevice. set true if null or property doesn't exist for backward-compatibility.
+                hasChanges = fixEscapedBooleanConfigParameters(oldConfiguration, NOTIFY_DEVICE_KEY, hasChanges, true);
+                // update sendAttributesUpdatedNotification.
+                hasChanges = fixEscapedBooleanConfigParameters(oldConfiguration, SEND_ATTRIBUTES_UPDATED_NOTIFICATION_KEY, hasChanges, false);
+                // update updateAttributesOnlyOnValueChange.
+                hasChanges = fixEscapedBooleanConfigParameters(oldConfiguration, UPDATE_ATTRIBUTES_ONLY_ON_VALUE_CHANGE_KEY, hasChanges, false);
```

Comment:
Basically defaultValue parameter in the fixEscapedBooleanConfigParameters method needs to be renamed to ifValueIsNull.

I have decided that if the rule node is broken and doesn't have any config set, then it's better to set the value to false to be in line with what we have done in the upgrade script from 3.5.1 to 3.6:

![image](https://github.com/thingsboard/thingsboard/assets/16356590/5aee989e-937d-411d-8597-ef48786d7842)

Imagine someone have a rule chain in the version control and there is no value at all(version 3.5.1 and below) so why we need to set it as true if in the upgrade from 3.5.1 we set it to false for existing nodes.

Reply:
Just realized that what I have mentioned in the comment above will never happen, since we will add this parameter with "false" value during upgrade from version 0.